balancer-js/src/modules/pools/pool-types/concerns/composableStable/exit.concern.ts

```diff
@@ -38,16 +171,282 @@ export class ComposableStablePoolExit implements ExitConcern {
     slippage,
     wrappedNativeAsset,
   }: ExitExactTokensOutParameters): ExitExactTokensOutAttributes => {
-    // TODO implementation
-    console.log(
-      exiter,
+    this.checkInputsExactTokensOut(tokensOut, amountsOut, pool);
+    const sortedValues = this.sortValuesExitExactTokensOut({
       pool,
-      tokensOut,
+      wrappedNativeAsset,
       amountsOut,
+      tokensOut,
+    });
+
+    const { bptIn, maxBPTIn } = this.calcBptInGivenExactTokensOut({
+      ...sortedValues,
       slippage,
-      wrappedNativeAsset
+    });
+
+    // Have to make sure amountsOut does not have value for BPT index
+    const userData = ComposableStablePoolEncoder.exitBPTInForExactTokensOut(
+      sortedValues.downscaledAmountsOutWithoutBpt,
+      maxBPTIn
     );
 
-    throw new Error('To be implemented');
+    const encodedData = this.encodeExitPool({
+      poolTokens: sortedValues.parsedTokens,
+      minAmountsOut: sortedValues.downscaledAmountsOutWithBpt,
+      userData,
+      exiter,
+      poolId: pool.id,
+    });
+
+    return {
+      ...encodedData,
+      maxBPTIn,
+      expectedBPTIn: bptIn,
+    };
+  };
+
+  /**
+   * Checks if the input of buildExitExactBPTIn is valid
+   */
+  checkInputsForSingleTokenExit = (
```

Comment:
I actually don't know! What are the pros/cons do you think?

Reply:
I think it helps to understand the scope of the function.
Is it only a helper method to be used locally or is it something that is worth having exposed to be used outside?
Anyway, I don't think it's critical - more like a nice to have. I wouldn't hold on merging the PR because of this.